swift/apple/FirezoneKit/Sources/FirezoneKit/Views/ResourceMenuSection.swift

```diff
@@ -0,0 +1,277 @@
+//
+//  ResourceMenuSection.swift
+//  (c) 2025 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+
+import SwiftUI
+
+/// Individual resource menu item with submenu containing resource details
+struct ResourceMenuItem: View {
+  let resource: Resource
+  @EnvironmentObject var store: Store
+
+  var body: some View {
+    Menu {
+      ResourceDetailsSubmenu(resource: resource)
+    } label: {
+      HStack(spacing: 4) {
+        if let symbolName = resourceSymbol {
+          Image(systemName: symbolName)
+        }
+        Text(resource.name)
+      }
+    }
+  }
+
+  var resourceSymbol: String? {
+    if resource.isInternetResource() {
+      return store.configuration.internetResourceEnabled ? "network" : "network.slash"
+    }
+    // For regular resources, show status indicator
+    switch resource.status {
+    case .online:
+      return "checkmark.circle.fill"
+    case .offline:
+      return "xmark.circle.fill"
+    case .unknown:
+      return "circle"
+    }
+  }
+}
+
+/// Submenu containing resource details, site info, and actions
+struct ResourceDetailsSubmenu: View {
+  let resource: Resource
+  @EnvironmentObject var store: Store
+
+  var body: some View {
+    Group {
+      // Address or description header
+      if resource.isInternetResource() {
+        Text("All network traffic")
+          .foregroundStyle(.secondary)
+
+        Divider()
+
+        Button(internetResourceToggleTitle) {
+          store.configuration.internetResourceEnabled.toggle()
+        }
+      } else {
+        // Show addressDescription if available, otherwise address
+        if let addressDescription = resource.addressDescription {
+          if let url = URL(string: addressDescription), url.host != nil {
+            Button {
+              Task { await NSWorkspace.shared.openAsync(url) }
+            } label: {
+              HStack {
+                Text(addressDescription)
+                Spacer()
+                Image(systemName: "arrow.up.forward")
+              }
+            }
+          } else {
+            Button(addressDescription) {
+              copyToClipboard(addressDescription)
+            }
+          }
+        } else if let address = resource.address {
+          Button(address) {
+            copyToClipboard(address)
+          }
+        }
+
+        Divider()
+
+        Text("Resource")
+          .foregroundStyle(.secondary)
+
+        Menu("Actions") {
```

Comment:
Apologies if I'm not understanding, but doesn't this create a submenu of `Actions` from the resource details menu? That would be 3 menus deep, which seems unnecessarily nested for the number of actions to perform.

Reply:
have you tried using it? IMHO it's way better, gets the not essential ways out of the way.

There's the alternative to just have actions inline, but it feel cluttery as 99% of time you just either want to open the thing or check it's status.

Reply:
What about the inlining that we talked about yesterday? `Resource` is now essentially just a single section with just another indirection to `Actions`. Maybe we can rename `Resource` to `Actions`?